dvc/repo/experiments/queue/local.py

```diff
@@ -124,22 +125,31 @@ def worker(self) -> "TemporaryWorker":
         )
 
     def spawn_worker(self):
-        from shortuuid import uuid
-
         from dvc_task.proc.process import ManagedProcess
 
         logger.debug("Spawning exp queue worker")
         wdir_hash = hashlib.sha256(self.wdir.encode("utf-8")).hexdigest()[:6]
-        node_name = f"dvc-exp-{wdir_hash}-1@localhost"
+        number = 1
+        node_name = f"dvc-exp-{wdir_hash}-{number}@localhost"
+        worker_status = self.active_worker()
+        while node_name in worker_status:
+            number += 1
+            node_name = f"dvc-exp-{wdir_hash}-{number}@localhost"
```

Comment:
The result `celery.control.inspect().active()` and `celery.control.ping()` returned is based on `node name`, different worker with the same node name will be regarded as the same.

Reply:
This is intentional. We want to make sure we are reusing node names (so for something like `queue start --jobs 4`, we would only use nodes 1 through 4 (and it would never increment the node number past 4).

When the dvc-task `TemporaryWorker` starts, it checks whether or not another worker with the same node name exists, and if it does exist, the newly started worker exits immediately. This ensures that we always have the correct number of workers running at a time.